onadata/libs/utils/logger_tools.py

```diff
@@ -275,6 +277,25 @@ def check_submission_permissions(request, xform):
               }))
 
 
+def check_submission_encryption(xform: XForm, xml: bytes) -> NoReturn:
+    """
+    Check that the submission is encrypted or unencrypted depending on the
+    encryption status of an XForm.
+
+    The submission is invalid if the XForm's encryption status is different
+    from the submissions
+    """
+    submission_element = ET.fromstring(xml)
+    encryption_status = submission_element.attrib.get('encrypted') == "yes"
```

Comment:
I wonder, is this the only attribute that we need to check to determine a submission is encrypted? Should we also confirm presence of or lack of `base64EncryptedKey` and `encryptedXmlFile`.

Reply:
All three [attributes are required](https://getodk.github.io/xforms-spec/encryption) in a properly built Encrypted submission; _Initially thought checking for just one would be okay. But, now I believe we can check if all three are present and also add functionality to determine whether the encrypted submission is valid_

I'll add these checks and re-request a review